code/game/objects/items/weapons/boomerang.dm

```diff
@@ -0,0 +1,90 @@
+/obj/item/weapon/boomerang
+	name = "boomerang"
+	desc = "A heavy, curved piece of wood used by Space Australians for hunting, sport, entertainment, cooking, religious rituals and warfare. When thrown, it will either deal a devastating blow to somebody's head, or return back to the thrower." //also used for shitposting
+	icon_state = "boomerang"
+
+	w_class = 3.0
+
+	throwforce = 25
+	throw_range = 7
+	throw_speed = 1
+
+	force = 7
+
+	autoignition_temperature = AUTOIGNITION_WOOD
+	fire_fuel = 3
+
+/obj/item/weapon/boomerang/Destroy()
+	..()
+
+	throwing = 0
+
+/obj/item/weapon/boomerang/toy
+	name = "toy boomerang"
+	desc = "A small plastic boomerang for children."
+
+	icon_state = "boomerang_toy"
+
+	w_class = 2.0
+
+	throwforce = 2
+	force = 1
+
+	starting_materials = list(MAT_PLASTIC = 1200)
+	melt_temperature = MELTPOINT_PLASTIC
+
+/obj/item/weapon/boomerang/throw_at(atom/target, range, speed, override = 1)
+	if(!usr)
+		return ..()
+
+	spawn()
+		while(src)
+			if(!throwing) //This stops the spinning if the item is no longer flying
+				animate(src, transform = null)
```

Comment:
there's no time frame for this ```animate()```, which means it happens instantly.
```animate()``` is 5x more expensive than a normal var assignment, so since this happens instantly, ```transform = null``` would be better.